aws-android-sdk-cognitoidentityprovider/src/main/java/com/amazonaws/mobileconnectors/cognitoidentityprovider/continuations/AuthenticationDetails.java

```diff
@@ -95,6 +95,7 @@ public AuthenticationDetails(String userId, Map<String, String> authenticationPa
         if (authenticationParameters != null) {
             this.authenticationType = CognitoServiceConstants.CHLG_TYPE_CUSTOM_CHALLENGE;
             this.authenticationParameters = authenticationParameters;
+            setAuthenticationParameter(CognitoServiceConstants.AUTH_PARAM_USERNAME, userId);
```

Comment:
Hey, I wasn't able to find where in the code later on this was needed - do you have it offhand? It seems like it shouldn't be needed since it stores userId directly.

Reply:
@TrekSoft  The http sigin in request fails without this parameter.
These parameters are going into the http request body further downstream in the code and AUTH_PARAM_USERNAME is the required parameter for the request to be successful because it is the **primary identifier**  in sign in request.  
**The other constructor in the same class is setting this parameter but this one is missing it. I am assuming it is ommitted by mistake.  If you look at the javadoc for this method, it already mentions the `userId`, but it is just missing in the code.** 

Success : Here is the http request body with the change included.
```
{
   "AuthFlow":"CUSTOM_AUTH",
   "AuthParameters":{
      **"USERNAME":"harshavsura26@gmail.com"**
   },
   "ClientMetadata":{
      
   },
   "ClientId":"c6rr3s4hpm0m82pehq6j973jq",
   "UserContextData":{
      "EncodedData":"eyJwYXlsb2FkIjoie1wiY29udGV4dERhdGFcIjp7XCJIYXNTaW1DYXJkXCI6XCJ0cnVlXCIsXCJB\ncHBsaWNhdGlvblZlcnNpb25cIjpcIjEuMFwiLFwiUGxhdGZvcm1cIjpcIkFORFJPSURcIixcIkFw\ncGxpY2F0aW9uVGFyZ2V0U2RrXCI6XCIzMFwiLFwiRGV2aWNlSWRcIjpcImEzOGExOTU5LWVjNmUt\nNDgyOC1hMzhlLWZjZGU4YzQwMDk1YToxNTk4NjQ1MzUxMTM0XCIsXCJQcm9kdWN0XCI6XCJjcm9z\nc2hhdGNoXCIsXCJTaW1Db3VudHJ5XCI6XCJ1c1wiLFwiQ2xpZW50VGltZXpvbmVcIjpcIi0wODow\nMFwiLFwiTmV0d29ya1R5cGVcIjpcIjBcIixcIlNpbU9wZXJhdG9yXCI6XCJTaW1wbGUgTW9iaWxl\nXCIsXCJEZXZpY2VTZGtWZXJzaW9uXCI6XCIyOVwiLFwiQnVpbGRUeXBlXCI6XCJ1c2VyXCIsXCJT\nY3JlZW5IZWlnaHRQaXhlbHNcIjpcIjI3MzNcIixcIkNhcnJpZXJcIjpcIlNpbXBsZSBNb2JpbGVc\nIixcIkRldmljZUZpbmdlcnByaW50XCI6XCJnb29nbGVcXFwvY3Jvc3NoYXRjaFxcXC9jcm9zc2hh\ndGNoOjEwXFxcL1FRM0EuMjAwODA1LjAwMVxcXC82NTc4MjEwOnVzZXJcXFwvcmVsZWFzZS1rZXlz\nXCIsXCJBcHBsaWNhdGlvbk5hbWVcIjpcIlN0cmFpZ2h0YXdheVwiLFwiRGV2aWNlQnJhbmRcIjpc\nImdvb2dsZVwiLFwiSXNOZXR3b3JrUm9hbWluZ1wiOlwiZmFsc2VcIixcIlRoaXJkUGFydHlEZXZp\nY2VJZFwiOlwiYW5kcm9pZF9pZFwiLFwiRGV2aWNlT3NSZWxlYXNlVmVyc2lvblwiOlwiMTBcIixc\nIkRldmljZUxhbmd1YWdlXCI6XCJlbl9VU1wiLFwiUGhvbmVUeXBlXCI6XCIxXCIsXCJTY3JlZW5X\naWR0aFBpeGVsc1wiOlwiMTQ0MFwiLFwiRGV2aWNlSGFyZHdhcmVcIjpcImNyb3NzaGF0Y2hcIixc\nIkRldmljZU5hbWVcIjpcIlBpeGVsIDMgWExcIn0sXCJ1c2VybmFtZVwiOlwiaGFyc2hhdnN1cmEy\nNkBnbWFpbC5jb21cIixcInVzZXJQb29sSWRcIjpcInVzLWVhc3QtMV85bzdyd3VaZjNcIixcInRp\nbWVzdGFtcFwiOlwiMTU5ODk4NDg1MjM2M1wifSIsInNpZ25hdHVyZSI6ImdRNkVpY0dMS1VJR1BL\nUzF2NzBWcUppeUtmYUZoSnZocGJwTTBiejl1a2c9XG4iLCJ2ZXJzaW9uIjoiQU5EUk9JRDIwMTcx\nMTE0In0=\n"
   }
}
```
Failure : Here is the http request body **with out** the change included.
```
{
   "AuthFlow":"CUSTOM_AUTH",
   "AuthParameters":{
      
   },
   "ClientMetadata":{
      
   },
   "ClientId":"c6rr3s4hpm0m82pehq6j973jq",
   "UserContextData":{
      "EncodedData":"eyJwYXlsb2FkIjoie1wiY29udGV4dERhdGFcIjp7XCJIYXNTaW1DYXJkXCI6XCJ0cnVlXCIsXCJB\ncHBsaWNhdGlvblZlcnNpb25cIjpcIjEuMFwiLFwiUGxhdGZvcm1cIjpcIkFORFJPSURcIixcIkFw\ncGxpY2F0aW9uVGFyZ2V0U2RrXCI6XCIzMFwiLFwiRGV2aWNlSWRcIjpcImEzOGExOTU5LWVjNmUt\nNDgyOC1hMzhlLWZjZGU4YzQwMDk1YToxNTk4NjQ1MzUxMTM0XCIsXCJQcm9kdWN0XCI6XCJjcm9z\nc2hhdGNoXCIsXCJTaW1Db3VudHJ5XCI6XCJ1c1wiLFwiQ2xpZW50VGltZXpvbmVcIjpcIi0wODow\nMFwiLFwiTmV0d29ya1R5cGVcIjpcIjBcIixcIlNpbU9wZXJhdG9yXCI6XCJTaW1wbGUgTW9iaWxl\nXCIsXCJEZXZpY2VTZGtWZXJzaW9uXCI6XCIyOVwiLFwiQnVpbGRUeXBlXCI6XCJ1c2VyXCIsXCJT\nY3JlZW5IZWlnaHRQaXhlbHNcIjpcIjI3MzNcIixcIkNhcnJpZXJcIjpcIlNpbXBsZSBNb2JpbGVc\nIixcIkRldmljZUZpbmdlcnByaW50XCI6XCJnb29nbGVcXFwvY3Jvc3NoYXRjaFxcXC9jcm9zc2hh\ndGNoOjEwXFxcL1FRM0EuMjAwODA1LjAwMVxcXC82NTc4MjEwOnVzZXJcXFwvcmVsZWFzZS1rZXlz\nXCIsXCJBcHBsaWNhdGlvbk5hbWVcIjpcIlN0cmFpZ2h0YXdheVwiLFwiRGV2aWNlQnJhbmRcIjpc\nImdvb2dsZVwiLFwiSXNOZXR3b3JrUm9hbWluZ1wiOlwiZmFsc2VcIixcIlRoaXJkUGFydHlEZXZp\nY2VJZFwiOlwiYW5kcm9pZF9pZFwiLFwiRGV2aWNlT3NSZWxlYXNlVmVyc2lvblwiOlwiMTBcIixc\nIkRldmljZUxhbmd1YWdlXCI6XCJlbl9VU1wiLFwiUGhvbmVUeXBlXCI6XCIxXCIsXCJTY3JlZW5X\naWR0aFBpeGVsc1wiOlwiMTQ0MFwiLFwiRGV2aWNlSGFyZHdhcmVcIjpcImNyb3NzaGF0Y2hcIixc\nIkRldmljZU5hbWVcIjpcIlBpeGVsIDMgWExcIn0sXCJ1c2VybmFtZVwiOlwiaGFyc2hhdnN1cmFA\nZ21haWwuY29tXCIsXCJ1c2VyUG9vbElkXCI6XCJ1cy1lYXN0LTFfOW83cnd1WmYzXCIsXCJ0aW1l\nc3RhbXBcIjpcIjE1OTg5ODU4NTMzNjdcIn0iLCJzaWduYXR1cmUiOiI2Uml5eFE4RnBCRVlqTzVG\nZE5iTFVQdmtINHBLcVwvUjZnQ2VSSjlKbUhBaz1cbiIsInZlcnNpb24iOiJBTkRST0lEMjAxNzEx\nMTQifQ==\n"
   }
}
```

Reply:
Awesome - thanks for that